File: app/src/main/java/com/example/javademogithubpractice/common/MyAppGlideModel.java


package com.example.javademogithubpractice.common;

import android.content.Context;

import com.bumptech.glide.Glide;
import com.bumptech.glide.GlideBuilder;
import com.bumptech.glide.Registry;
import com.bumptech.glide.annotation.GlideModule;
import com.bumptech.glide.load.engine.cache.InternalCacheDiskCacheFactory;
import com.bumptech.glide.module.AppGlideModule;
import com.bumptech.glide.request.RequestOptions;
import com.example.javademogithubpractice.AppConfig;
import com.example.javademogithubpractice.R;

@GlideModule
public class MyAppGlideModel extends AppGlideModule {

    @Override
    public boolean isManifestParsingEnabled() {
        return super.isManifestParsingEnabled();
    }

    @Override
    public void applyOptions(Context context, GlideBuilder builder) {
        super.applyOptions(context, builder);
        builder.setDiskCache(new InternalCacheDiskCacheFactory(context, AppConfig.IMAGE_MAX_CACHE_SIZE));
        RequestOptions requestOptions = RequestOptions.placeholderOf(R.drawable.logoeasyhub);
        builder.setDefaultRequestOptions(requestOptions);
    }

    @Override
    public void registerComponents(Context context, Glide glide, Registry registry) {
        super.registerComponents(context, glide, registry);
    }

}

File: KotlinDemoGithubProject/app/src/main/java/com/example/kotlindemogithubproject/common/MyAppGlideModel.kt
package com.example.kotlindemogithubproject.common

import android.content.Context
import com.bumptech.glide.Glide
import com.bumptech.glide.GlideBuilder
import com.bumptech.glide.Registry
import com.bumptech.glide.annotation.GlideModule
import com.bumptech.glide.load.engine.cache.InternalCacheDiskCacheFactory
import com.bumptech.glide.module.AppGlideModule
import com.bumptech.glide.request.RequestOptions
import com.example.kotlindemogithubproject.AppConfig
import com.example.kotlindemogithubproject.R


@GlideModule
class MyAppGlideModel : AppGlideModule() {

    override fun isManifestParsingEnabled(): Boolean {
        return super.isManifestParsingEnabled()
    }

    override fun applyOptions(context: Context?, builder: GlideBuilder?) {
        super.applyOptions(context, builder)
        builder!!.setDiskCache(InternalCacheDiskCacheFactory(context, AppConfig.IMAGE_MAX_CACHE_SIZE))
        val requestOptions = RequestOptions.placeholderOf(R.mipmap.logo)
        builder.setDefaultRequestOptions(requestOptions)
    }

    override fun registerComponents(context: Context?, glide: Glide?, registry: Registry?) {
        super.registerComponents(context, glide, registry)
    }

}
